donkeycar/parts/perfmon.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Performance monitor for analyzing real-time CPU/mem/execution frequency
+
+author: @miro (Meir Tseitlin) 2020
+
+Note:
+"""
+import time
+import psutil
+
+
+class PerfMonitor:
+
+    def __init__(self, cfg):
+
+        self.STATS_BUFFER_SIZE = 10
+        self._calc_buffer = [cfg.DRIVE_LOOP_HZ for i in range(self.STATS_BUFFER_SIZE)]
+        self._runs_counter = 0
+        self._last_calc_time = time.time()
+        print("Performance monitor activated.")
+
+    def run(self):
```

Comment:
Let's make this a threaded part. Look at the camera or the OLED part for more information.

Reply:
Ok. Will do...

Reply:
Moved heavy functions into thread